core/commonTest/src/contract/list/ImmutableListTest.kt

```diff
@@ -25,7 +25,7 @@ class ImmutableListTest {
 
         assertFailsWith<NoSuchElementException> { empty1.iterator().next() }
 
-        compareLists(emptyList(), empty1)
+        compareLists(emptyList<Int>(), empty1)
```

Comment:
AFAIK, this won't be a problem in K2 soon

Reply:
Warning will remain. See [KT-66196](https://youtrack.jetbrains.com/issue/KT-66196).

Reply:
But let's decide if we want to take this into the main branch.

Reply:
Given that the change is needed for K2, I don't have objections to merging it into `master`. This also aligns with the goal of keeping the `kotlin-community/dev` branch close to `master`.